### PR TITLE
lightning: split and scatter regions in batches

### PR DESCRIPTION
**Pull request by gozssky**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/33618

Problem Summary:

Region scatter may timeout when split and scatter many regions at a time.

### What is changed and how it works?

Add a batch limit for split and scatter regions. In a batch, lightning processes up to 4096 ranges. After all ranges in batch have been processed completely, lightning can process the next batch.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)

I prepared about 1.8T data for a table which contains only one clustered primary key. TiDB cluster have ten TiKV nodes, each node have 200% cpu limit and 16GiB memory limit. Then I used lightning to import these data to TiDB cluster and observed the scatter operator status in grafana dashboard. To reproduce huge amount of regions, I set `region-split-size` to 8MiB. Finally, about 260k regions were created.

Master
image
Many scatter operator is timeout.

This PR
image

There is no operator timeout after this PR.

- [ ] No code

### Release note

```release-note
lightning: split and scatter regions in batches
```

### CODE REVIEW

File: br/pkg/lightning/backend/local/localhelper.go [hunk 1]
```diff
@@ -428,9 +450,15 @@ func (local *local) waitForScatterRegion(ctx context.Context, regionInfo *split.
...
 		if err != nil {
-			log.L().Warn("scatter region failed: do not have the region",
-				logutil.Region(regionInfo.Region))
-			return
+			log.L().Warn(
+				"wait for scatter region failed, will scatter again",
+				logutil.Region(regionInfo.Region),
+				zap.Error(err),
+			)
+			if err := local.splitCli.ScatterRegion(ctx, regionInfo); err != nil {
```
Comment (D3Hunter) [inline br/pkg/lightning/backend/local/localhelper.go:458]: maybe we can retry scatter only on retryable error

**Review by D3Hunter — COMMENTED** [review]
rest lgtm

File: br/pkg/lightning/backend/local/localhelper.go (hunk 1 above)
Comment (gozssky) [reply br/pkg/lightning/backend/local/localhelper.go]: Fixed in 851ada81af9165c63f2e54e28e6b60eb30e217a8.

File: br/pkg/lightning/backend/local/localhelper.go [hunk 2]
```diff
@@ -471,8 +466,8 @@ func (local *local) waitForScatterRegion(ctx context.Context, regionInfo *split.
 	}
 }
 
-func (local *local) isScatterRegionFinished(ctx context.Context, regionID uint64) (bool, error) {
-	resp, err := local.splitCli.GetOperator(ctx, regionID)
+func (local *local) isScatterRegionFinished(ctx context.Context, regionInfo *split.RegionInfo) (bool, error) {
```
Comment (D3Hunter) [inline br/pkg/lightning/backend/local/localhelper.go:469]: maybe rename this func to reflect it retry scatter inside

Comment (gozssky) [reply br/pkg/lightning/backend/local/localhelper.go:469]: Is it ok to rename it to `checkScatterRegionFinsihedOrReScatter`?

**Review by D3Hunter — APPROVED** [review]
LGTM

**Review by lichunzhu — APPROVED** [review]
LGTM

**Pull request merged**